### DISCUSSION TITLE
Completion with Linol

### DISCUSSION THREAD

**Discussion by SoilihBenSoilih** [c-cube/linol#7] [General]
Hello, 

Hope you're doing well !
I'm writing an Lsp server using Linol, and it's working fine.
Diagnostics are here. But I have some problems with completion, 
I don't really understand what the argument of method `on_req_completion`
represents. The context for exemple, so i have some problems to implement
it. 
Have you done an exemple in a project, so i can refer there or can you show me
an exemple if it's not the case with statics data for exemple?

Thank you a lot !!!

**Comment by c-cube** [comment]
Hi, you can take a look at https://github.com/c-cube/trustee/blob/master/src/lsp/trustee_lsp.ml#L161-L211 for example. The idea is that you take a uri + position, and return a list of completions for that point in the file.

**Reply by SoilihBenSoilih** [reply]
Hello,

Thank you a lot !!!
